### PR TITLE
Isaac/pipeline

### LINKED ISSUE #59: Bug: Models not being saved
I'm trying to use pickle to save models and it's entirely ineffective. It should work, because scikit suggests it. Some other people have had the same issue.

### LINKED ISSUE #63: Fixing Tests: Integrating Splitter
We're trying to integrate Ryan's splitter with the rest of the project and replace the older splitting functionality with it. Currently running into issues with the R code running properly (it is likely failing out).

### PR DESCRIPTION
**Pull request by ASAAR**
I finally fixed the issue with the models, I don't know what was happening, but it's working now. I did have to remove the AdaBoost algorithm because it refused to work. Integrating the new splitter has also run into some issues that we'll be fixing soon. Just realize that not all tests pass currently.

This is mostly a bug fix,

Closes #59 
Closes #63 

For me the rest of this iteration and the next one is going to be focused on testing, refining the framework, and using design patterns to make it cleaner.

@asclines @RyanMcBerg Please review

### CODE REVIEW

**Comment by ASAAR** [comment]
@asclines I updated this with some more tests. But the more I look at the system, the more I think I want to integrate the "train" and "test" mode, or at least make them more distinct. Because to some degree, the logging system will replace "train" mode. Which isn't even named well.

**Comment by ASAAR** [comment]
@asclines @RyanMcBerg @ZakeryFyke @telelu03 

Anyone want to review?

File: modules/toolbox/r_split.R [hunk 1]
```diff
@@ -6,14 +6,15 @@
...
+setwd("datasets") ### need to set dynamically
 
 library(dplyr)
 
 Args <- commandArgs(trailingOnly =  T)
 
 ### reads in to csv and makes all String variables Factors #####
 
-dataset <- read.csv(as.character(Args), stringsAsFactors = T)
+dataset <- read.csv(as.character(Args), sep = ';', stringsAsFactors = T)
```
Comment (asclines) [inline modules/toolbox/r_split.R]: Is the separator always going to be ';'?

File: tests/test_merge.py [hunk 2]
```diff
@@ -20,4 +20,4 @@ def test_merge(self):
         data_test = pd.read_csv("full.dataset.csv")
 
         # Assert
-        self.assertEqual(data_set, data_test, msg="Test Successful")
\ No newline at end of file
+        self.assertEqual(data_set, data_test, msg="Test Successful")
```
Comment (asclines) [inline tests/test_merge.py]: Doesn't that message only show upon error?

File: README.md [hunk 3]
```diff
@@ -25,3 +27,14 @@ Software Engineering 2 class kaggle competition repo. We'll also be building a b
 
 ###### Coding Standards (Please Call Us On It)
 - [Python](https://google.github.io/styleguide/pyguide.html)
+
+###### Running The Framework
+We will be streamlining this over future iterations, but currently it works like this. The framework is also not fully featured and only a few algorithms are fully supported currently.
+- run "python launcher.py"
+- provide answers to the following questions,
+  - train or test: train means that you just want to see what kinds of results you might get, test means it will both train the model and run it against test data. Test is not fully supported yet.
+  - training file name: In either case, this is necessary. Please include the file ending (i.e. .csv).
```
Comment (ASAAR) [inline README.md]: Just assumed, however once we setup config we will be adding path info as well.

File: modules/toolbox/r_split.R (hunk 1 above)
Comment (ASAAR) [inline modules/toolbox/r_split.R]: @RyanMcBerg is fixing that

File: tests/test_merge.py (hunk 2 above)
Comment (ASAAR) [inline tests/test_merge.py]: fixed

**Pull request merged**